src/MPDevice.cpp

```diff
@@ -49,7 +49,8 @@ MPDevice::MPDevice(QObject *parent):
 
             if (isBLE())
             {
-                bleImpl->readUserSettings(pMesProt->getPayloadBytes(data, 2, 4));
+                if(pMesProt->getMessageSize(data) >= 4) // these settings are sent only when user is logged in
+                    bleImpl->readUserSettings(pMesProt->getPayloadBytes(data, 2, 4));
```

Comment:
Awesome catch...

moreover, they are now included in the get_status message... so you we may just remove that.... what do you think @deXol ?

Reply:
this is the MPCmd::MOOLTIPASS_STATUS command's callback. In line 53 it is reading the user settings from get_status.

Reply:
oops, my bad!